### PR TITLE
Implemented exponentiation

### PR DESCRIPTION
**Pull request by g0lc0de**
- Implements #2 

Team number 2 includes @nikohass @maxomnia @Dahmspiegel  @g0lc0de , sowie Dustin ohne Github Account

### CODE REVIEW

**Comment by g0lc0de** [comment]
All tests run through

File: src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java [hunk 1]
```diff
@@ -332,29 +335,83 @@ AstNode signedTerm3( ){
...
 	public AstNode exponentiation( ){
-		// TODO Auto-generated method stub
-		return super.exponentiation( );
+		// Rule: Exponentiation: atomicExpression Exponent
+		// Select Set: LPAREN, IDENTIFIER, NUMERIC_LITERAL, EOF
+
+		if (LPAREN == LEXER.lookahead().TYPE
+				|| IDENTIFIER == LEXER.lookahead().TYPE
+				|| NUMERIC_LITERAL == LEXER.lookahead().TYPE
+				|| EOF == LEXER.lookahead().TYPE) {
```
Comment (marco-haupt) [inline src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java:360]: Why did you include EOF here?

File: src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java [hunk 2]
```diff
@@ -332,29 +335,83 @@ AstNode signedTerm3( ){
...
+			case LESS:
+			case GREATER_EQUAL:
+			case LESS_EQUAL:
+			case EQUAL:
+			case NOT_EQUAL:
+			case LAND:
+			case LOR:
+			case RPAREN:
+			case MODULO:
+			case EOF:
```
Comment (marco-haupt) [inline src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java:397]: I don't think that's quite right, but obviously I'm assuming just these tokens in the test 🤔 Looks like I made a mistake here... Please forgive. I'll merge your change anyway so we can talk about it on Tuesday.

**Review by marco-haupt — APPROVED** [review]
The selection sets for the production rules don't seem to be correct, but I think the fault is mine. At least all the tests are running through successfully. I'm merging your PR so we can talk about it on Tuesday. Thank you for your contribution! 🙂

File: src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java (hunk 1 above)
Comment (g0lc0de) [reply src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java:394]: When EOF is not included, one tests does not pass. This is why we added it to the  check.
image

**Pull request merged**